### PR TITLE
Adding in Y24

### LINKED ISSUE #223: New THEMIS grain model
It would be great if the THEMIS 2.0 dust grain model (Ysard et al, 2024) can be added:
https://www.ias.u-psud.fr/themis/downloads_2.html

### PR DESCRIPTION
**Pull request by karllark**
Add in Ysard et al. (2024) grain model curve.

Closes #223.

### CODE REVIEW

File: dust_extinction/grain_models.py [hunk 1]
```diff
@@ -606,3 +606,85 @@ def __init__(self, modelname="MWRV31", **kwargs):
         self.data_tolerance = 1e-6
 
         super().__init__(**kwargs)
+
+
+class Y24(BaseExtGrainModel):
+    r"""
+    Ysard et al. (2024) Grain Model
```
Comment (mdecleir) [inline dust_extinction/grain_models.py:613]: Might be worth mentioning that this is THEMIS 2.0, to avoid confusion in the future.

**Review by mdecleir — COMMENTED** [review]
Looks good!
Just one comment, as we discussed before, you need to make sure all models have a different color. I often use matplotlib's tab20 color map (https://matplotlib.org/stable/users/explain/colors/colormaps.html) if I need more than 10 colors.

**Comment by karllark** [comment]
> Looks good! Just one comment, as we discussed before, you need to make sure all models have a different color. I often use matplotlib's tab20 color map (https://matplotlib.org/stable/users/explain/colors/colormaps.html) if I need more than 10 colors.

Can you share how you do this?

**Comment by mdecleir** [comment]
Yes, first you create a color map:

`colors = plt.get_cmap("tab20")`

Then, you can use that map in your for loop, and loop through the 20 colors in that colormap with the modulo operator:

`for i,model in enumerate(models):
plt.scatter(model[x],model[y],color=colors%i)`

**Comment by karllark** [comment]
Updated based on comments.

**Pull request merged**